modules/osd-monitoring-troubleshooting-issues.adoc

```diff
@@ -0,0 +1,110 @@
+
+// Module included in the following assemblies:
+//
+// * monitoring/osd-troubleshooting-monitoring-issues.adoc
+
+[id="troubleshooting-monitoring-issues_{context}"]
+= Determining why user-defined project metrics are unavailable
+
+If metrics are not displaying when monitoring user-defined projects, follow these steps to troubleshoot the issue.
+
+.Procedure
+
+. Query the metric name and verify that the project is correct:
+.. From the *Developer* perspective in the OpenShift Container Platform web console, select *Monitoring* -> *Metrics*.
+.. Select the project that you want to view metrics for in the *Project:* list.
+.. Choose a query from the *Select Query* list, or run a custom PromQL query by selecting *Show PromQL*.
++
+The *Select Query* pane shows the metric names.
++
+Queries must be done on a per-project basis. The metrics that are shown relate to the project that you have selected.
+. Verify that the pod that you want metrics from is actively serving metrics. Run the following `oc exec` command into a pod to target the `podIP`, `port`, and `/metrics`.
++
+[source,terminal]
+----
+$ oc exec <sample_pod> -n <sample_namespace> -- curl <target_pod_IP>:<port>/metrics
+----
++
+[NOTE]
+====
+You must run the command on a pod that has `curl` installed.
+====
++
+The following example output shows a result with a valid version metric.
++
+.Example output
+[source,terminal]
+----
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+# HELP version Version information about this binary-- --:--:-- --:--:--     0
+# TYPE version gauge
+version{version="v0.1.0"} 1
+100   102  100   102    0     0  51000      0 --:--:-- --:--:-- --:--:-- 51000
+----
++
+An invalid output indicates that there is a problem with the corresponding application.
+
+. If you are using `PodMonitor` CRD, verify that `PodMonitor` CRD is configured to point to the correct pods using label matching. For more information, see the Prometheus Operator documentation.
+. If you are using ServiceMonitor, and if the `/metrics` endpoint of the pod is showing metric data, follow these steps to verify the configuration:
```

Comment:
Should this be "If you are using a `ServiceMonitor` CRD"?

Reply:
Yes, I checked with Taylor Fahlman and he said that either is correct, it's semantics, but I did change it.